File: 2021/src/test/java/de/jfriemel/aoc/days/Day13Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day13Test {

    private static final String INPUT_STRING = """
            6,10
            0,14
            9,10
            0,3
            10,4
            4,11
            6,0
            6,12
            4,1
            0,13
            10,12
            3,4
            3,0
            8,4
            1,10
            2,14
            8,10
            9,0
                        
            fold along y=7
            fold along x=5""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day13().part1(INPUT);
        assertEquals("17", result);
    }

    @Test
    public void testPart2() {
        String result = new Day13().part2(INPUT);
        assertEquals("""
                
                \u2588\u2588\u2588\u2588\u2588
                \u2588   \u2588
                \u2588   \u2588
                \u2588   \u2588
                \u2588\u2588\u2588\u2588\u2588
                    \s
                    \s""", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day13Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day13Test {
    private val input = """
        #.##..##.
        ..#.##.#.
        ##......#
        ##......#
        ..#.##.#.
        ..##..##.
        #.#.##.#.
        
        #...##..#
        #....#..#
        ..##..###
        #####.##.
        #####.##.
        ..##..###
        #....#..#
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day13.part1(input)
        Assertions.assertEquals("405", result)
    }

    @Test
    fun testPart2() {
        val result = Day13.part2(input)
        Assertions.assertEquals("400", result)
    }
}
